content/docs/tutorials/cloud-tutorial.md

```diff
@@ -0,0 +1,562 @@
+`timekeeper-connector.sh` to make the distinction easier to see.
+
+%copy first-line%
+```bash
+$ ./timekeeper-connector.sh "test123"
+topic "timekeeper-with-connector" created
+
+```
+
+Nothing much is returned when running, but now the database will be more
+interesting. Try checking its contents with `fluvio consume`!
+
+You can stop the connector by deleting it.
+
+%copy first-line%
+```bash
+$ fluvio connector delete cat-facts
+connector "cat-facts" deleted
+
+```
+
+Wait, now the Fluvio instance is getting cluttered! Sure the cat facts are nice,
+but now there is JSON mixed in with the timestamped notes!
+
+Fear not, we have *SmartModules* to help with that.
+
+## SmartModules
+
+SmartModules are user defined functions set to run on and modify the inputs/outputs to
+a Fluvio database.
+
+Want to filter so that only JSON records that match a specific priority tag are
+recorded? A Filter SmartModule can be written to only let through records with
+specific values.
+
+You create a SmartModule by using Rust and generating it based on the SmartModule
+template available [at the github repository](https://github.com/infinyon/fluvio-smartmodule-template).
+
+### Making a SmartModule
+
+If we want to clean up the timekeeper records and make it so that the messages
+are marked separately from the JSON objects, we will need a SmartModule.
+Specifically we will need a map SmartModule.
+
+We need to go through some setup steps though.
+
+First, check if `wasm32` is listed by `rustup` as an installed target.
+
+%copy first-line%
+```bash
+$ rustup target list | grep installed
+wasm32-unknown-unknown (installed)
+x86_64-unknown-linux-gnu (installed)
+```
+
+-> If it is not installed, run `$ rustup target add wasm32-unknown-unknown` to install it.
+
+Next, install `cargo-generate`, this may take a minute or two.
+
+%copy first-line%
+```bash
+$ cargo install cargo-generate
+```
+
+Now you can download the template with `cargo-generate`.
+
+-> You will have to fill in a couple of required arguments below, prefaced by the "🤷" emoji.
+
+%copy first-line%
+```bash
+cargo generate --git https://github.com/infinyon/fluvio-smartmodule-template
+🤷   Project Name : catfacts-map
+🔧   Destination: /home/[...]/projects/[...]/catfacts-map ...
+🔧   Generating template ...
+✔ 🤷   Want to use SmartModule parameters? · true
+✔ 🤷   Which type of SmartModule would you like? · map
+[1/7]   Done: .cargo/config.toml
+[2/7]   Done: .cargo
+[3/7]   Done: .gitignore
+[4/7]   Done: Cargo.toml
+[5/7]   Done: README.md
+[6/7]   Done: src/lib.rs
+[7/7]   Done: src
+🔧   Moving generated files into: `/home/[...]/projects/[...]/catfacts-map`...
+💡   Initializing a fresh Git repository
+✨   Done! New project created /home/[...]/projects/[...]/catfacts-map
+
+```
+
+There should now be a new directory labeled `catfacts-map` in your working directory.
+
+Now we can edit the `catfacts-map/src/lib.rs` file to get what we need!
+
+%copy%
+```rust
```

Comment:
Can you add the Rust source code, and just embed the `lib.rs` file?

In the future, want to be able to ensure that this code continues to be in a buildable state, with Dependabot making sure it stays up to date over fluvio releases.